Delete adjacent space in remaining cases - Part 3

zuphilip opened the pull request:
( Reference to #1301 )

adam3smith commented on documents-d-archeologie-francaise.csl:
```diff
@@ -595,22 +595,26 @@
+              <text macro="page"/>
+              <text macro="isbn"/>
+              <text macro="url"/>
+              <text macro="archive" prefix=" (" suffix=")"/>
```
this is an extra space, too, right? remove.

adam3smith commented on harvard-kings-college-london.csl:
```diff
@@ -250,8 +250,8 @@
         <else-if type="chapter paper-conference" match="any">
-          <text macro="title" prefix=" '" suffix="', "/>
           <group prefix=" " delimiter=" ">
+            <text macro="title" prefix="'" suffix="',"/>
```
could you change this to quotes="true" (the style is already en-GB, so they'll be single quotes with comma outside)

zuphilip replied:
If I understand correctly that would then be
```
<text macro="title" quotes="true" suffix=","/>
```
Is this correct?

adam3smith replied:
yes

adam3smith commented on norma-portuguesa-405.csl:
```diff
@@ -309,8 +309,8 @@
           <date-part name="month" form="short" suffix=". " text-case="lowercase"/>
```
while we're at it, the parentheses should really be set as a prefix/suffix on the ```date``` in case any of the ```date-part``` are missing.

adam3smith commented on philosophia-scientiae.csl:
```diff
@@ -280,9 +280,9 @@
-          <group suffix="; ">
-            <number variable="number-of-volumes" form="numeric" suffix=" "/>
-            <text term="volume" form="short" prefix=" " suffix=" " plural="true"/>
+          <group suffix=" ; " delimiter=" ">
```
That space there isn't actually right given the styleguide:
     

> Boltzmann, Ludwig [1909], Wissenschaftliche Abhandlungen von Ludwig Boltzmann, 3 volumes; New York: Chelsea Publishing Company, 1968.

adam3smith replied:
also, delete the form="short" as per the same example.

adam3smith commented on proceedings-of-the-royal-society-b.csl:
```diff
@@ -49,10 +49,8 @@
   </macro>
   <macro name="publisher">
```
given how popular this style is, we should also do the publisher/publisher place thing right, i.e. with a group with delimiter=": "

adam3smith commented on revue-archeologique-du-centre-de-la-france.csl:
```diff
@@ -277,7 +277,7 @@
           <if type="thesis">
             <text variable="genre" suffix=". "/>
-            <text macro="publisher" prefix=" "/>
+            <text macro="publisher"/>
```
while I was looking at this style -- we should run a sed script to remove all empty affixes (i.e. prefix="" and suffix="") Since those never do anything, that can just be scripted and separate. @rmzelle might even consider including it in his general maintenance script?

adam3smith commented on risk-analysis.csl:
```diff
@@ -102,17 +102,17 @@
-            <group suffix=".">
-              <text macro="publisher" prefix=" "/>
-              <group suffix="." prefix=", ">
+            <text variable="container-title" text-case="title" suffix="."/>
```
this must be suffix=". "--no way the volume (or publisher or whatever) starts after a period without a space.

adam3smith commented on standards-in-genomic-sciences.csl:
```diff
@@ -182,7 +182,7 @@
+            <group delimiter=";">
               <date variable="issued">
                 <date-part name="year"/>
               </date>
```
another thing we could consider removing via script: groups with a single element. Sed probably isn't a good choice here, but python should work nicely. cc @rmzelle

zuphilip replied:
Yeah, I had the same thought. But this can become more tricky if the only child node has children itself. Moreover, AFAIK one can use `group` also for a conditional existence statement.

adam3smith replied:
> But this can become more tricky if the only child node has children itself.

Does it though? I can't think of a case where that'd make a difference. Obviously if the group has affixes, those need to be passed down to the child.

> Moreover, AFAIK one can use group also for a conditional existence statement.

But only in a meaningful way if there are more than one child items, right? The conditional is that if you have a term and a variable in a group, the term only gets printed when the variable does. Or what did you have in mind?

zuphilip replied:
1) Finding the relevant cases might be a different. I would guess that for that you need to go trough the XML structure and looking for group nodes with only one child. (Sure, this is doable, but it is different from find&replace of a regexp).
2) You are right. That was a false thought from my side...
3) One would have to merge all attributes from the group and its child node. I am not sure if this is possible automatically.

adam3smith replied:
right, that's why I was thinking python, which has a nice XML parser, which Rintze uses for the re-indent/reorder script anyway. I agree a regex would probably not be feasible, at least not reasonably so.

adam3smith commented on the-british-journal-of-sociology.csl:
```diff
@@ -167,9 +167,9 @@
-          <group suffix="', ">
-            <text macro="title" prefix=" '"/>
-            <text macro="editor" prefix=" "/>
+          <group prefix=" '" suffix="', " delimiter=" ">
```
this won't typically actually occur, but obviously we don't want quotes around the editor.
I'd like the quotes="true" to be handled to the title macro (for the else case in l. 83) and then removed from the rest of the style.

adam3smith commented on wheaton-college-phd-in-biblical-and-theological-studies.csl:
```diff
@@ -331,8 +333,10 @@
-            <text variable="volume" prefix=" " suffix=" of "/>
+            <group delimiter=" ">
+              <text term="volume" form="short"/>
+              <text variable="volume" suffix=" of "/>
```
@dowens76 -- could you take a quick look at this? This looks like number-of-volumes is missing here?

dowens76 replied:
I am no longer studying there, but based on the current style guide at http://www.wheaton.edu/~/media/Files/Academics/Departments/BTS/PhD%20Program/Style_Guide.pdf and my memory, the only place you include the number of volumes is when you are citing a work as a whole or in the bibliography. But just by looking at the commit I cannot tell if this creates a problem or not. Here is an example of how things should be: 

Note: G. E. Ladd, “Kingdom of God,” in vol 3 of *The International Standard Bible Encyclopedia*, ed. Geoffrey W. Bromiley, rev. ed. (Grand Rapids: Eerdmans, 1986), 29.
Bibliography: G. E. Ladd, “Kingdom of God.” In vol 3 of *The International Standard Bible Encyclopedia*, edited by Geoffrey W. Bromiley, 23–39. Rev. ed. Grand Rapids: Eerdmans, 1986. 

But this commit does not change that behavior, right?

zuphilip replied:
No, the commit didn't change that behaviour and it seems to conform with your examples: vol 3 of <i>Title</i>. I think @adam3smith has tought it could also be e.g. vol 3 of 6 (then the 6 = number of volumes would have been missing).

adam3smith replied:
yup, exactly. This is fine as is then.

adam3smith commented:
OK, sorry that most of these notes are actually fixes to things that were wrong before you did this, but I figured while I was reviewing this I might as well catch obvious errors. Let me know if that got too much, I can take some of them.

zuphilip commented:
I updated the commit. This should contain all your comments and I added another style: theory-culture-and-society.csl , for which I had first do look at the documentation. I appreciate the possibility to improve on some styles and I think such search pattern can lead us to the more problematic structured implementations. It is worth to spend some time to improve on them. So, no problem.

adam3smith commented:
man, those new Travis builds are fast @rmzelle @inukshuk --thanks again for this.
Looks great now, thanks.
